tests/components/knx/test_config_flow.py

```diff
@@ -950,12 +950,12 @@ async def test_configure_secure_knxkeys_invalid_signature(hass: HomeAssistant):
         assert secure_knxkeys["errors"][CONF_KNX_KNXKEY_PASSWORD] == "invalid_signature"
 
 
+@patch("homeassistant.components.knx.async_setup_entry", AsyncMock(return_value=True))
```

Comment:
```suggestion
@patch("homeassistant.components.knx.async_setup_entry", return_value=True)
```

Reply:
Copied that from ZHA. I did try your suggestion before, but it didn't work.
`mock_config_entry.add_to_hass(hass)` raised `AttributeError: 'coroutine' object has no attribute 'append'`

Reply:
I think you need to add a parameter in the test signature for the decorator mock. Otherwise it will be confused.

I recommend not mixing pytest fixtures and patch decorators. I'd extract this patch to a single pytest fixture that is autoused.